### PR TITLE
Add support for filtering the rules list

### PR DESCRIPTION
**Pull request by j-f1**
~This PR mostly works (and addresses something that’s been bothering me for a while)! The sticking point is that it can’t properly pull in the names of payees/schedules/categories so they aren’t searchable. Not sure how to fix that without making the app super laggy.~

Searches happen essentially based on all of the text displayed in the table.

### CODE REVIEW

**Comment by Kidglove57** [comment]
Just a thought - ignore if out of order! How straightforward would it be to include filtering the schedules list within this work? My schedules list extends to two pages and it is hard to find things.

**Comment by j-f1** [comment]
That would be a different PR, probably.

**Comment by j-f1** [comment]
Ok, I think I’ve got it working!

File: packages/desktop-client/src/components/ManageRules.js [hunk 1]
```diff
@@ -222,6 +229,14 @@ export function ConditionExpression({
   );
 }
 
+function describeSchedule(s, payee) {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/ManageRules.js:232]: **🥜 nitpick:** `s` is very vague‏
```suggestion
function describeSchedule(schedule, payee) {
```

File: packages/desktop-client/src/components/ManageRules.js [hunk 2]
```diff
@@ -495,15 +505,82 @@ function RulesList({
...
+
+  let { data: schedules } = SchedulesQuery.useQuery();
+  let filterData = useSelector(state => ({
+    payees: state.queries.payees,
+    categories: state.queries.categories.list,
+    accounts: state.queries.accounts,
+    schedules
+  }));
+
+  console.log('fil', filterData);
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/ManageRules.js:571]: **🔨 warning:** ‏oops

**Review by MatissJanis — APPROVED** [review]
Just two small nitpicks, but overall LGTM!

**Review by MatissJanis — APPROVED** [review]

**Comment by shall0pass** [comment]
I encountered this error when trying to use the Rules search box. 

TypeError: Cannot read properties of undefined (reading 'name')
    at $r (ManageRules.js:514:42)
    at ManageRules.js:536:9
    at Array.flatMap (<anonymous>)
    at ManageRules.js:530:30
    at ManageRules.js:576:13
    at Array.filter (<anonymous>)
    at ManageRules.js:575:17
    at Object.ya [as useMemo] (react-dom.production.min.js:3830:1)
    at t.useMemo (react.production.min.js:397:1)
    at Kr (ManageRules.js:571:23)

**Comment by j-f1** [comment]
Weird! Would you be able to drop a debugger breakpoint at that line and get some more context about why it’s breaking? Looking at the code, it seems like the category for the rule could not be found — if you’re able to `console.log` out the contents of the rule (at `ManageRules.js:536:9`), that would be super helpful for debugging!

**Comment by shall0pass** [comment]
You were right!  I had a weird rule in my list pointing to a category that looked more like a random identifier.  Anyways, I deleted the weird rule and the search box works fine.

**Comment by shall0pass** [comment]
Follow up. The reason the rule was in a word state is because the rule existed and later the category was deleted causing the Id string to be shown.
Screenshot_20230203-052619-792

**Pull request merged**